addon/components/validated-input/template.hbs

```diff
@@ -0,0 +1,25 @@
+<div class="form-group">
+    {{#if isTextArea}}
+        {{textarea value=value type=type placeholder=placeholder class="form-control" name=valuePath}}
+    {{/if}}
+
+    {{#if isTextField}}
+        {{input type=type value=value placeholder=placeholder class="form-control" name=valuePath}}
+    {{/if}}
```

Comment:
To make sure we never have more than one input field rendered, can we change this to `else if` (also including isDateField below). It should also fall back to an `input` if none of the flags are set, so something like:
```
{{#if isTextArea}}
    {{textarea ...}}
{{else if isDateField}}
    {{bootstrap-datepicker ...}
{{else}}
    {{input ...}}
{{/if}}
```
This will ensure there is always exactly one input field rendered. You can then also drop the `isTextField` computed property and not worry about initializing `inputType`.